lib/iris/tests/unit/io/test_save.py

```diff
@@ -18,6 +18,7 @@
 
 class TestSave(tests.IrisTest):
     def test_pathlib_save(self):
+        # DOESN'T CHECK TILDES WORK DUE TO MOCK.
```

Comment:
@pp-mo do you have any opinions on what should happen here? Not testing expansion feels like an unfortunate oversight but it sounds like any alternative would be a big project?

Reply:
I think that limitation **_is_** ok, because here we are not testing that part of the behaviour.  Instead, we are just checking that the different valid types of argument are all processed OK.
The expansion behaviour can be tested elsewhere.

On the other hand,  I'm not convinced it actually is tested.  I think there should really be a testcase  in test_expand_filespecs to do that.  
See [other comment there](https://github.com/SciTools/iris/pull/4913/files#r951477688)